test/view_component/base_test.rb

```diff
@@ -90,4 +90,20 @@ def test_does_not_render_additional_newline_with_render_in
   ensure
     ActionView::Template::Handlers::ERB.strip_trailing_newlines = false if Rails::VERSION::MAJOR >= 7
   end
+
+  def test_evaled_component
+    source = <<~RUBY
+      class EvaledComponent < ViewComponent::Base
+        def initialize(cta: nil, title:)
+          @cta = cta
+          @title = title
+        end
+        private
+
+        attr_reader :cta, :title
+      end
+    RUBY
+
+    eval(source)
```

Comment:
Looks like we need to add a rubocop exception for this line for our linters to pass ❤️